src/plugins/jma/datasource.py

```
@@ -0,0 +1,396 @@
+
+
+logger = logging.getLogger(__name__)
+
+
+class JMADataSource(DataSource):
+    def __init__(
+        self,
+        api_client: Optional[RateLimitedAPIClient] = None,
+        cache: Optional[Cache] = None,
+        metrics: Optional[MetricsCollector] = None,
+        retry_policy: Optional[Any] = None,
```

Ruff (0.12.2)

Unused method argument: `retry_policy`

(ARG002)